backend/controller/console.go

```diff
@@ -392,3 +410,35 @@ func eventDALToProto(event dal.Event) *pbconsole.Event {
 		panic(fmt.Errorf("unknown event type %T", event))
 	}
 }
+
+func graph(sch *schema.Schema) (map[string][]string, error) {
+	out := make(map[string][]string)
+	for _, module := range sch.Modules {
+		err := buildGraph(sch, module, out)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return out, nil
+}
+
+// buildGraph recursively builds the dependency graph
+func buildGraph(sch *schema.Schema, module *schema.Module, out map[string][]string) error {
+	out[module.Name] = module.Imports()
+	for _, dep := range module.Imports() {
+		var depModule *schema.Module
+		for _, m := range sch.Modules {
+			if m.String() == dep {
+				depModule = m
+				break
+			}
+		}
+		if depModule != nil {
+			err := buildGraph(sch, module, out)
```

Comment:
Oh yeah 🤦 good call :)

I was wondering if it would be useful to have a `Graph` method on `schema` to allow us to get a graph of any schema anywhere vs. here in `console`. Maybe it could be used by the `buildengine` as well

Reply:
Do you mean move `buildengine.Engine.Graph()` there? Also what is the difference between this `buildGraph` and the one in `Engine`?

Reply:
Yeah, they are very similar and can hopefully be consolidated. I think the primary difference is how they each get their `deps` in the `buildengine` uses `project.Config().Dependecies` or `sch.Imports()` and this console version always uses `Imports()`